Replace custom test renderer with gotestsum

jakule opened the pull request:
Why
----
test render has a few bugs that we don't have time to fix. Instead of maintaining our own solution, we can use something that already works.

What
----

All instances of the custom test renderer (`${RENDER_TESTS}`) in the Makefile have been replaced with gotestsum calls. This allows to provide a more standardized and extendible tool for handling the test output. The gotestsum tool, being a popular and actively maintained project, adds additional features such as output formatting options and compatibility with developer tools. As a result, it improves the readability and accessibility of our test results.

Notes
---

This PR does half the job. I was not able to replace `test-render` in all places as `gotestsum` is missing on our Docker image. I added `gotestsum` to the Docker image and devbox, so `test-render` can be removed after the Docker image is updated, but I also decided to use `gotestsum` in a few places, so the output can be compared between the old an new tooling.
I added conditional `gotestsum` installation to the makefile, as, IMO, people who don't have is installed should just get the latest version automatically instead of figuring out how to do it.

jakule committed:
Replace custom test renderer with gotestsum

All instances of the custom test renderer (`${RENDER_TESTS}`) in the Makefile have been replaced with gotestsum calls. This allows to provide a more standardized and extendible tool for handling the test output. The gotestsum tool, being a popular and actively maintained project, adds additional features such as output formatting options and compatibility with developer tools. As a result, it improves the readability and accessibility of our test results.

jakule committed:
Add gotestsum and update Makefile, Dockerfile, and devbox files

Changes have been made in Makefile, Dockerfile, and devbox.json files to install gotestsum if it's not already installed and use it in the test commands. The devbox.lock file has also been updated accordingly to reflect the addition of gotestsum.

jakule committed:
Update gotestsum installation in Makefile

The Makefile has been adjusted to ignore any errors while installing 'gotestsum'. This is a temporary measure until 'gotestsum' is added to the Dockerfile, to prevent the build process from failing.

capnspacehook reviewed (approved):
LGTM, excited to see this fully integrated soon!

jakule committed:
Fix devbox?

espadolini commented on Makefile:
```diff
@@ -643,6 +643,10 @@ TOOLINGDIR := ${abspath ./build.assets/tooling}
 $(RENDER_TESTS): $(wildcard $(TOOLINGDIR)/cmd/render-tests/*.go)
 	cd $(TOOLINGDIR) && go build -o "$@" ./cmd/render-tests
+# Install gotestsum if it's not already installed
+ ifeq (, $(shell which gotestsum))
```
```suggestion
 ifeq (, $(shell command -v gotestsum))
```

jakule committed:
Apply a review suggestion
Revert devbox changes - devbox seems to be broken. I'll introduce this change in a new PR, so the current one is not blocked on an admin approval.

jakule committed:
Revert `make e2e-aws` to use render-test to prevent CI failing
